static/app/views/alerts/rules/metric/details/metricHistory.tsx

```diff
@@ -1,145 +1,54 @@
-          <StatusIndicator
-            status={isCritical ? 'error' : 'warning'}
-            tooltipTitle={t('Status: %s', isCritical ? t('Critical') : t('Warning'))}
-          />
-        )}
-        <Link
-          to={{
-            pathname: alertDetailsLink(organization, incident),
-            query: {alert: incident.identifier},
-          }}
-        >
-          #{incident.identifier}
-        </Link>
-      </Cell>
-      <Cell>
-        {incident.alertRule.comparisonDelta ? (
-          <Fragment>
-            {alertName} {curentTrigger?.alertThreshold}%
-            {t(
-              ' %s in %s compared to the ',
-              incident.alertRule.thresholdType === AlertRuleThresholdType.ABOVE
-                ? t('higher')
-                : t('lower'),
-              timeWindow
-            )}
-            {COMPARISON_DELTA_OPTIONS.find(
-              ({value}) => value === incident.alertRule.comparisonDelta
-            )?.label ?? COMPARISON_DELTA_OPTIONS[0].label}
-          </Fragment>
-        ) : (
-          <Fragment>
-            {alertName}{' '}
-            {incident.alertRule.thresholdType === AlertRuleThresholdType.ABOVE
-              ? t('above')
-              : t('below')}{' '}
-            {curentTrigger?.alertThreshold} {t('in')} {timeWindow}
-          </Fragment>
-        )}
-      </Cell>
-      <Cell>
-        {activityDuration &&
-          getDynamicText({
-            value: <Duration abbreviation seconds={activityDuration / 1000} />,
-            fixed: '30s',
-          })}
-      </Cell>
-      <Cell>
-        <StyledDateTime
-          date={getDynamicText({
-            value: incident.dateCreated,
-            fixed: 'Mar 4, 2022 10:44:13 AM UTC',
-          })}
-          year
-          seconds
-          timeZone
-        />
-      </Cell>
-    </Fragment>
-  );
-}
-
 type Props = {
+  activations?: AlertRuleActivation[];
   incidents?: Incident[];
 };
 
-function MetricHistory({incidents}: Props) {
+function MetricHistory({incidents, activations}: Props) {
   const organization = useOrganization();
   const filteredIncidents = (incidents ?? []).filter(
     incident => incident.activities?.length
   );
   const numOfIncidents = filteredIncidents.length;
 
+  const activationTriggers: ActivationTriggerActivity[] = [];
+  activations?.forEach(activation => {
+    activationTriggers.push({
+      type: ActivationTrigger.ACTIVATED,
+      activator: activation.activator,
+      conditionType: activation.conditionType,
+      dateCreated: activation.dateCreated,
+    });
+    if (activation.isComplete) {
+      activationTriggers.push({
+        type: ActivationTrigger.FINISHED,
+        activator: activation.activator,
+        conditionType: activation.conditionType,
+        dateCreated: activation.finishedAt,
+      });
+    }
+  });
+
+  const sortedActivity = [...filteredIncidents, ...activationTriggers].sort((a, b) =>
+    a.dateCreated > b.dateCreated ? -1 : 1
+  );
```

Comment:
might be nice to pull this out as a helper function, `const sortedActivity = getSortedActivities(activities)`, then you could memoize that response.

Reply:
🤷 could also just `useMemo` anyways still right?
I'm not quite as familiar with the space tradeoff here but i figured sorting a list isn't _that_ heavy of an operation (as opposed to querying data or pulling from a store or something) and may not be worth memoizing.

Reply:
the reason i'd recommend useMemo is because this will execute every render. Looking at the data and how many props are being passed into this means there's a greater possibility of a rerender event.

Reply:
Eh yea 🤷 

I'll just memoize then